### PR TITLE
Add listen/rpc "prefer_ipv6" options to DNS lookup

### LINKED ISSUE #4751: scylla.yaml parameter "rpc_interface_prefer_ipv6: false" is ignored
Scylla version (or git commit hash): master

As far as I can see, `rpc_interface_prefer_ipv6` parameter is ignored.

In Apache Cassandra, the same parameter is read
https://github.com/apache/cassandra/blob/cassandra-3.11/src/java/org/apache/cassandra/config/DatabaseDescriptor.java#L857

### PR DESCRIPTION
**Pull request by elcallio**
Add listen/rpc "prefer_ipv6" options to DNS lookup of bind addresses for API/rpc/prometheus etc .

Fixes #4751
    
Adds using a preferred address family to dns name lookups related to
listen address and rpc address, adhering to the respective "prefer" options.
    
API, prometheus and broadcast address are all considered to be covered by
the "listen_interface_prefer_ipv6" option.
    
Note: scylla does not yet support actual interface binding, but these
options should apply equally to address name parameters.
    
Setting a "prefer_ipv6" option automtially enables ipv6 dns family query.

### CODE REVIEW

**Comment by avikivity** [comment]
Pull requests don't absolve you from writing a good subject line and description (these go into the merge commit).

**Comment by elcallio** [comment]
Sorry, it felt pretty self-explanatory, but you are right. Blurb and commit messages enriched.

File: gms/inet_address.cc [hunk 1]
```diff
@@ -44,8 +44,13 @@
...
-future<gms::inet_address> gms::inet_address::lookup(sstring name, opt_family family) {
-    return seastar::net::dns::resolve_name(name, family).then([](seastar::net::inet_address&& a) {
-        return make_ready_future<gms::inet_address>(a);
+future<gms::inet_address> gms::inet_address::lookup(sstring name, opt_family family, opt_family preferred) {
+    return seastar::net::dns::get_host_by_name(name, family).then([preferred](seastar::net::hostent&& h) {
+        for (auto& addr : h.addr_list) {
+            if (!preferred || addr.in_family() == preferred) {
+                return gms::inet_address(addr);
+            }
+        }
```
Comment (avikivity) [inline gms/inet_address.cc]: I know someone who would have coded this with boost::partition()

**Comment by avikivity** [comment]
Queued as ce2b0b26822b5da073af649f0145635e77198fb3

**Pull request merged**